staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -99,10 +99,9 @@ func getBaseEnv() (*cel.Env, error) {
 //  - nil Program, non-nil Error: Compilation resulted in an error
 //  - nil Program, nil Error: The provided rule was empty so compilation was not attempted
 // perCallLimit was added for testing purpose only. Callers should always use const PerCallLimit as input.
-func Compile(s *schema.Structural, isResourceRoot bool, perCallLimit uint64) ([]CompilationResult, error) {
+func Compile(s *schema.Structural, isResourceRoot bool, declType *celmodel.DeclType, perCallLimit uint64) ([]CompilationResult, error) {
```

Comment:
`isResourceRoot` param looks unused now and could disagree with the construction of the `declType` param... drop the unused param?

also, document `declType` must correspond to `s`?

Reply:
Yes, dropping isResourceRoot here.  I'll add the godoc in line with the other godoc being added about declType.